<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>812</width>
    <height>681</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <widget class="QPushButton" name="pushButton_optimize">
    <property name="geometry">
     <rect>
      <x>610</x>
      <y>480</y>
      <width>113</width>
      <height>32</height>
     </rect>
    </property>
    <property name="text">
     <string>Optimize</string>
    </property>
   </widget>
   <widget class="QCustomPlot" name="plot" native="true">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>40</y>
      <width>521</width>
      <height>511</height>
     </rect>
    </property>
   </widget>
   <widget class="QWidget" name="layoutWidget">
    <property name="geometry">
     <rect>
      <x>559</x>
      <y>10</y>
      <width>221</width>
      <height>461</height>
     </rect>
    </property>
    <layout class="QVBoxLayout" name="verticalLayout_9">
     <item>
      <layout class="QVBoxLayout" name="verticalLayout">
       <item>
        <widget class="QLabel" name="label">
         <property name="text">
          <string>Function</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QComboBox" name="comboBox_func">
         <item>
          <property name="text">
           <string>f(x, y) = x^2 + y^2</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>f(x, y) = y^2 + cos(x + y) + 1</string>
          </property>
         </item>
        </widget>
       </item>
       <item>
        <widget class="QPushButton" name="pushButton_plot">
         <property name="text">
          <string>Plot</string>
         </property>
        </widget>
       </item>
      </layout>
     </item>
     <item>
      <layout class="QVBoxLayout" name="verticalLayout_2">
       <item>
        <widget class="QLabel" name="label_2">
         <property name="text">
          <string>Optimization method</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QComboBox" name="comboBox_method">
         <item>
          <property name="text">
           <string>Gradient descent</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>Random search</string>
          </property>
         </item>
        </widget>
       </item>
      </layout>
     </item>
     <item>
      <layout class="QVBoxLayout" name="verticalLayout_4">
       <item>
        <widget class="QLabel" name="label_3">
         <property name="text">
          <string>Area</string>
         </property>
        </widget>
       </item>
       <item>
        <layout class="QVBoxLayout" name="verticalLayout_3">
         <item>
          <layout class="QHBoxLayout" name="horizontalLayout_2">
           <item>
            <widget class="QLineEdit" name="lineEdit_xleft">
             <property name="text">
              <string>-10</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QLabel" name="label_4">
             <property name="text">
              <string>&lt; x &lt;</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QLineEdit" name="lineEdit_xright">
             <property name="text">
              <string>10</string>
             </property>
            </widget>
           </item>
          </layout>
         </item>
         <item>
          <layout class="QHBoxLayout" name="horizontalLayout">
           <item>
            <widget class="QLineEdit" name="lineEdit_yleft">
             <property name="text">
              <string>-10</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QLabel" name="label_5">
             <property name="text">
              <string>&lt; y &lt; </string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QLineEdit" name="lineEdit_yright">
             <property name="text">
              <string>10</string>
             </property>
            </widget>
           </item>
          </layout>
         </item>
         <item>
          <layout class="QVBoxLayout" name="verticalLayout_10">
           <item>
            <widget class="QLabel" name="label_6">
             <property name="text">
              <string>Starting point</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QTabWidget" name="tabWidget">
             <property name="currentIndex">
              <number>0</number>
             </property>
             <widget class="QWidget" name="tab">
              <attribute name="title">
               <string>By value</string>
              </attribute>
              <widget class="QWidget" name="">
               <property name="geometry">
                <rect>
                 <x>20</x>
                 <y>0</y>
                 <width>169</width>
                 <height>49</height>
                </rect>
               </property>
               <layout class="QVBoxLayout" name="verticalLayout_7">
                <item>
                 <layout class="QHBoxLayout" name="horizontalLayout_3">
                  <item>
                   <layout class="QVBoxLayout" name="verticalLayout_5">
                    <item>
                     <widget class="QLabel" name="label_7">
                      <property name="text">
                       <string>x_0</string>
                      </property>
                     </widget>
                    </item>
                    <item>
                     <widget class="QLineEdit" name="lineEdit_x0">
                      <property name="text">
                       <string>2</string>
                      </property>
                     </widget>
                    </item>
                   </layout>
                  </item>
                  <item>
                   <layout class="QVBoxLayout" name="verticalLayout_6">
                    <item>
                     <widget class="QLabel" name="label_8">
                      <property name="text">
                       <string>y_0</string>
                      </property>
                     </widget>
                    </item>
                    <item>
                     <widget class="QLineEdit" name="lineEdit_y0">
                      <property name="text">
                       <string>8</string>
                      </property>
                     </widget>
                    </item>
                   </layout>
                  </item>
                 </layout>
                </item>
               </layout>
              </widget>
             </widget>
             <widget class="QWidget" name="tab_2">
              <attribute name="title">
               <string>By click</string>
              </attribute>
              <widget class="QLabel" name="label_x0y0">
               <property name="geometry">
                <rect>
                 <x>10</x>
                 <y>10</y>
                 <width>181</width>
                 <height>51</height>
                </rect>
               </property>
               <property name="text">
                <string>Click on the graph to select 
a starting point</string>
               </property>
              </widget>
             </widget>
            </widget>
           </item>
          </layout>
         </item>
        </layout>
       </item>
      </layout>
     </item>
     <item>
      <layout class="QVBoxLayout" name="verticalLayout_8">
       <item>
        <widget class="QLabel" name="label_9">
         <property name="text">
          <string>Stop crierion</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QComboBox" name="comboBox_stop">
         <item>
          <property name="text">
           <string>f-epsilon</string>
          </property>
         </item>
        </widget>
       </item>
      </layout>
     </item>
    </layout>
   </widget>
   <widget class="QLabel" name="label_consoleResult">
    <property name="geometry">
     <rect>
      <x>570</x>
      <y>520</y>
      <width>191</width>
      <height>111</height>
     </rect>
    </property>
    <property name="text">
     <string/>
    </property>
   </widget>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>812</width>
     <height>22</height>
    </rect>
   </property>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
 </widget>
 <customwidgets>
  <customwidget>
   <class>QCustomPlot</class>
   <extends>QWidget</extends>
   <header location="global">qcustomplot.h</header>
   <container>1</container>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections/>
</ui>
